lib/logstash/outputs/webhdfs.rb

```diff
@@ -0,0 +1,282 @@
+  # supports any string and can include %{name} and other dynamic
+  # strings.
+  #
+  # If this setting is omitted, the full json representation of the
+  # event will be written as a single line.
+  config :message_format, :validate => :string
+
+  # Sending data to webhdfs in x seconds intervals.
+  config :idle_flush_time, :validate => :number, :default => 1
+
+  # Sending data to webhdfs if event count is above, even if store_interval_in_secs is not reached.
+  config :flush_size, :validate => :number, :default => 500
+
+  # WebHdfs open timeout, default 30s (in ruby net/http).
+  config :open_timeout, :validate => :number, :default => 30
+
+  # The WebHdfs read timeout, default 30s (in ruby net/http).
+  config :read_timeout, :validate => :number, :default => 30
+
+  # Use httpfs mode if set to true, else webhdfs.
+  config :use_httpfs, :validate => :boolean, :default => false
+
+  # Retry some known webhdfs errors. These may be caused by race conditions when appending to same file, etc.
+  config :retry_known_errors, :validate => :boolean, :default => true
+
+  # How long should we wait between retries.
+  config :retry_interval, :validate => :number, :default => 0.5
+
+  # How many times should we retry.
+  config :retry_times, :validate => :number, :default => 5
+
+  # Compress output. One of ['none', 'snappy', 'gzip']
+  config :compression, :validate => ["none", "snappy", "gzip"], :default => "none"
+
+  # Set snappy chunksize. Only neccessary for stream format. Defaults to 32k. Max is 65536
+  # @see http://code.google.com/p/snappy/source/browse/trunk/framing_format.txt
+  config :snappy_bufsize, :validate => :number, :default => 32768
+
+  # Set snappy format. One of "stream", "file". Set to stream to be hive compatible.
+  config :snappy_format, :validate => ["stream", "file"], :default => "stream"
+
+  # Remove @timestamp field. Hive does not like a leading "@", but we need @timestamp for path calculation.
+  config :remove_at_timestamp, :validate => :boolean, :default => true
+
+  public
+
+  def register
+    begin
+      require 'webhdfs'
+    rescue LoadError
+      @logger.error("Module webhdfs could not be loaded.")
+    end
+    if @compression == "gzip"
+      begin
+        require "zlib"
+      rescue LoadError
+        @logger.error("Gzip compression selected but zlib module could not be loaded.")
+      end
+    elsif @compression == "snappy"
+      begin
+        require "snappy"
+      rescue LoadError
+        @logger.error("Snappy compression selected but snappy module could not be loaded.")
+      end
+    end
+    @files = {}
+    @host, @port = @server.split(':')
+    @client = prepare_client(@host, @port, @user)
+    # Test client connection.
+    begin
+      @client.list('/')
+    rescue => e
+      @logger.error("Webhdfs check request failed. (namenode: #{@client.host}:#{@client.port}, Exception: #{e.message})")
+    end
+    buffer_initialize(
```

Comment:
Indention corrected in https://github.com/dstore-dbap/logstash-output-webhdfs-1/commit/6f31d176a7c9270d58a42ecd599f5b277049b232